pkgs/top-level/all-packages.nix

```
@@ -11652,9 +11652,7 @@ with pkgs;
 
   pystring = callPackage ../development/libraries/pystring { };
 
-  raysession = python3Packages.callPackage ../applications/audio/raysession {
-    inherit (qt5) qttools;
-  };
+  raysession = python3Packages.callPackage ../applications/audio/raysession {};
```

```suggestion
  raysession = python3Packages.callPackage ../applications/audio/raysession {
    inherit (qt5) qttools wrapQtAppsHook;
  };
```
I think it is a good idea to add `wrapQtAppsHook` here.

---

I don't get it: what's the advantage of this compared to doing `qt5.wrapQtAppHook` as I do now? For me it is just harder to test as I need to remember to supply to good arguments to callPackage, and I only saw a single other package (by the same author) importing qttools this way. Am I missing something?

---

Not only `qttools`, but there are other examples like `qmake`, `qtbase`, `wrapQtAppsHook`, etc. It is also commonly done in other namespaces, not just `qt5`.
The advantage is that you can separate qt version changes from application changes. For example, `qutebrowser`.

On the other hand, it does not seem to be that much of an advantage for this application, but I do not think it is necessary to change it actively.

---

https://github.com/NixOS/nixpkgs/issues/204303